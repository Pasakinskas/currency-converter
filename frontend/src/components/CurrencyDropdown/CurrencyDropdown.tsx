import React, { ChangeEvent } from 'react';

type currencyDropdownProps = {
  id: string;
  className?: string;
  currencyCodes: Array<string>;
  onChange: (event: ChangeEvent<HTMLSelectElement>) => void;
  labelText: string
}

export function CurrencyDropdown(props: currencyDropdownProps) {
  const { 
    id, 
    currencyCodes,
    labelText,
    onChange: handleChange
  } = props;

  return (
    <React.Fragment>
      <label htmlFor={id}>{labelText}</label>
      <select id={id} className="pure-u-1-5" onChange={handleChange}>
        {currencyCodes
          .sort()
          .map(code => <option key={code}>{code}</option>)
        }
      </select>
    </React.Fragment>
  )
}
