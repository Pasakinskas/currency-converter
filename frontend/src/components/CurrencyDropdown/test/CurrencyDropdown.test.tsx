import React from "react";
import TestRenderer from 'react-test-renderer';
import { CurrencyDropdown } from "../CurrencyDropdown";
import { mount } from "enzyme";

describe("currencyDropdown", () => {
  test("Renders correctly", () => {
    const testDropdown = <CurrencyDropdown 
      id="test-id"
      labelText="mock-dropdown"
      onChange={e => e}
      currencyCodes={["EUR", "USD", "GBP"]}
      />
    const dropdown = TestRenderer
      .create(testDropdown)
      .toJSON();
 
    expect(dropdown).toMatchSnapshot();
  });

  test("displays initial to-dos", () => {
    const testDropdown = <CurrencyDropdown 
    id="test-id"
    labelText="mock-dropdown"
    onChange={e => e}
    currencyCodes={["EUR", "USD", "GBP"]}
    />
    const wrapper = mount(testDropdown);
    expect(wrapper.find("option")).toHaveLength(4);
  });
});