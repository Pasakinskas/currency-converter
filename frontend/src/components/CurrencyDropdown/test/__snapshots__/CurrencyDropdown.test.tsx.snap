// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`currencyDropdown Renders correctly 1`] = `
Array [
  <label
    htmlFor="test-id"
  >
    mock-dropdown
  </label>,
  <select
    className="pure-u-1-5"
    defaultValue="DEFAULT"
    id="test-id"
    onChange={[Function]}
  >
    <option
      disabled={true}
      hidden={true}
      value="DEFAULT"
    >
      -
    </option>
    <option>
      EUR
    </option>
    <option>
      GBP
    </option>
    <option>
      USD
    </option>
  </select>,
]
`;
